ext/windows/service/daemon.rb

```diff
@@ -127,23 +128,16 @@ def log(msg, level)
         File.open(LOG_FILE, 'a:UTF-8') { |f| f.puts("#{Time.now} Puppet (#{level}): #{msg}") }
       end
 
-      case level
-        when :debug, :info, :notice
-          report_windows_event(EVENTLOG_INFORMATION_TYPE,0x01,msg.to_s)
-        when :err, :alert, :emerg, :crit
-          report_windows_event(EVENTLOG_ERROR_TYPE,0x03,msg.to_s)
-        else
-          report_windows_event(EVENTLOG_WARNING_TYPE,0x02,msg.to_s)
-      end
+      native_type, native_id = Puppet::Util::Windows::EventLog.to_native(level)
+      report_windows_event(native_type, native_id, msg.to_s)
     end
   end
 
   def report_windows_event(type,id,message)
     begin
       eventlog = nil
-      eventlog = Win32::EventLog.open("Application")
+      eventlog = Puppet::Util::Windows::EventLog.open("Puppet")
```

Comment:
Looping back on this one, strictly speaking, the change to `Puppet` from `Application` was incorrect.

`RegisterEventSource` looks under the `EventLog` registry key, and will use the name given *ELSE* it will fall back to `Application` which is known to always exist.  So what we've changed here is that Puppet will use a `Puppet` event log should it exist, but gracefully falls back to `Application` otherwise.

In practical terms, this doesn't cause any problems as the old behavior is preserved. However, it also has the bonus feature of accidentally enabling logging to a different log than `Application`, which is a long-standing feature request for end users.  So this ends up being a dark feature, which solves https://tickets.puppetlabs.com/browse/PUP-5068 and is easily turned on by end users without any additional changes to Puppet or packaging.  So this turns out to be an incredibly happy accident!

I validated against agent package 5.1.0 the following:

* created a new 2012R2 host
* ran the PowerShell code `New-EventLog -Source Puppet -LogName Puppet -MessageResource C:\Program Files\Puppet Labs\Puppet\bin\puppetres.dll` to create a new log named `Puppet` with `Puppet` as a valid source (this creates the underlying event log file at `C:\Windows\System32\winevt\Logs\Puppet.evtx`)
* installed the agent - which registers the source `Puppet` under `Application`
* opened up Event Viewer to take a look at the new `Puppet` log, where all of the messages are now going to instead of `Application`

What makes this particular solution great is that the call to `New-EventLog` can easily be put into a PowerShell based `exec` resource. While initial Puppet messages will still be in the `Application` log, anything after the resource has been applied will go into the `Puppet` log instead:

![screen shot 2017-08-24 at 11 49 35 am](https://user-images.githubusercontent.com/394081/29683148-331eb4d4-88c2-11e7-9592-c2fe1105a05e.png)